chore(doc-gen): include type parameters in d.ts file

petebacondarwin opened the pull request:
Closes #2859

petebacondarwin committed:
chore(doc-gen): include type parameters in d.ts file

Closes #2859

mhevery commented on docs/dgeni-package/templates/type-definition.template.html:
```diff
@@ -5,6 +6,19 @@
 {%- endmacro -%}
+
+
+{%- macro memberInfo(member, level) -%}
```
This change has messed up indentation, and I can't seem to fix it. :-(

petebacondarwin replied:
fixed

mhevery commented on docs/dgeni-package/templates/type-definition.template.html:
```diff
@@ -26,25 +40,14 @@
-    new ({% for param in export.newMember.parameters %}{$ param $}{% if not loop.last %}, {% endif %}{% endfor %})
-    {%- if export.newMember.returnType %}: {$ export.newMember.returnType $}{%- else -%}: any{% endif -%}
-    ;
+  {%- if export.newMember %}
```
The code now renders `__new()` instead of `new ()`

petebacondarwin replied:
fixed

mhevery commented on docs/dgeni-package/templates/type-definition.template.html:
```diff
@@ -26,25 +40,14 @@
-    {%- if member.returnType %}: {$ member.returnType $}{%- else -%}: any{% endif -%}
-    ;
+  {%- if export.callMember %}
+    {$ memberInfo(export.callMember, 5) $}
```
The code now renders `__call()` instead of `()`

petebacondarwin replied:
fixed

petebacondarwin committed:
chore(doc-gen): include type parameters in d.ts file

Closes #2859

petebacondarwin committed:
chore(doc-gen): stop crash in `gulp public_docs`

This change prevents the doc generation from crashing due to a missing module,
but perhaps we need to think of a way of being able to generate the angular.d.ts
file correctly in `gulp public_docs`, perhaps by removing the need for
angular2/angular2.api

mhevery commented:
Merged: 0e945e465d7df01556c84a2bc511f9cd0489d1f3
